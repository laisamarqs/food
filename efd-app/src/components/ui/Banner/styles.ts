import styled from 'styled-components'
import { breakpoints } from '../../../styles'

export const BannerContainer = styled.div`
  position: relative;
  height: 17.5rem;

  img {
    height: 100%;
    width: 100%;
    object-fit: cover;
    filter: brightness(50%);
  }
`

export const Titles = styled.div`
  z-index: 1;
  display: grid;
  color: #fff;
  font-size: 2rem;
  position: absolute;
  top: 25px;
  left: 170px;

  span {
    font-weight: 100;
  }

  h2 {
    margin-top: 156px;
    font-weight: 900;
    font-size: 32px;
  }

  @media only screen and (max-width: ${breakpoints.tablet}) {
    padding-left: 2.7rem;
    left: 0;
  }

  @media only screen and (max-width: ${breakpoints.mobile}) {
    font-size: 1.3rem;
    left: 25px;
    padding: 0;

    h2 {
      margin-top: 130px;
    }
  }
`