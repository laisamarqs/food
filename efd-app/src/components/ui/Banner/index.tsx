import Image from '../../../utils/Image'
import * as Style from './styles'

export type BannerProps = {
  image: string
  type: string
  title: string
}

const Banner = ({ image, type, title }: BannerProps) => (
  <Style.BannerContainer>
    <Image src={image} alt={title} />
    <Style.Titles>
      <span>{type}</span>
      <h2>{title}</h2>
    </Style.Titles>
  </Style.BannerContainer>
)

export default Banner