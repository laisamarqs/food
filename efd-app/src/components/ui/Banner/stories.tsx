import type { Meta } from '@storybook/react'

import Banner, { BannerProps } from '.'

const meta = {
  title: 'ui/Banner',
  component: Banner,
  parameters: {
    layout: 'fullscreen'
  },
  tags: ['autodocs'],
  args: {
    image:
      'https://upload.wikimedia.org/wikipedia/commons/1/1e/Comida_de_mercado.jpg',
    title: 'La Doce Vita',
    type: 'Gourmet'
  }
} satisfies Meta<typeof Banner>

// eslint-disable-next-line react-refresh/only-export-components
export default meta

export const Template = (args: BannerProps) => {
  return <Banner {...args} />
}