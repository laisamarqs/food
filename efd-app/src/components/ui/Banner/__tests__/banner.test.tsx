import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import Banner from '..'
import { expect } from 'vitest'

const props = {
  image: 'img/img1.png',
  title: 'restaurante',
  type: 'italiana'
}

describe('<Banner />', () => {
  it('should be render correctly', () => {
    const { container } = render(<Banner {...props} />)

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute(
      'src',
      props.image
    )
    expect(
      screen.getByRole('heading', { name: props.title })
    ).toBeInTheDocument()
    expect(screen.getByText(props.type)).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})