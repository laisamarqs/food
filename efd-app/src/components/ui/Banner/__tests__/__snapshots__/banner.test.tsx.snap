// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Banner /> > should be render correctly 1`] = `
<div
  class="sc-dcJsrY gzPucS"
>
  <img
    alt="restaurante"
    class="loading"
    height="100"
    loading="lazy"
    src="img/img1.png"
    width="100"
  />
  <div
    class="sc-iGgWBj ccBhfb"
  >
    <span>
      italiana
    </span>
    <h2>
      restaurante
    </h2>
  </div>
</div>
`;